bob-apps/bin/brt.rs

```diff
@@ -299,45 +624,170 @@ fn validate_bytes(a: &[u8], checksum: u32) -> AnyResult<()> {
     Ok(())
 }
 
-struct Settings {
+enum Settings {
+    Recovery(RecoverySettings),
+    Validate(ValidateBlobSettings),
+    ValidateIndex(ValidateIndexSettings),
+}
+
+struct RecoverySettings {
     input: String,
     output: String,
     validate_every: usize,
 }
 
+struct ValidateBlobSettings {
+    path: String,
+    blob_suffix: String,
+    backup_suffix: String,
+    validate_every: usize,
+    fix: bool,
+    confirm: bool,
+}
+
+struct ValidateIndexSettings {
+    path: String,
+    index_suffix: String,
+    delete: bool,
+    confirm: bool,
+}
+
 impl Settings {
     fn from_matches() -> AnyResult<Settings> {
         let matches = get_matches();
-        Ok(Settings {
-            input: matches
-                .value_of(INPUT_BLOB_OPT)
-                .expect("Required")
-                .to_string(),
-            output: matches
-                .value_of(OUTPUT_BLOB_OPT)
-                .expect("Required")
-                .to_string(),
-            validate_every: matches
-                .value_of(VALIDATE_EVERY_OPT)
-                .expect("Has default")
-                .parse()?,
-        })
+        match matches.subcommand() {
+            (RECOVERY_COMMAND, Some(matches)) => Ok(Settings::Recovery(RecoverySettings {
```

Comment:
Probably it's better to move each branch matches parsing to the corresponding structure constructor

Reply:
Done